api/client/utils_test.go

```diff
@@ -25,9 +24,7 @@ func defaultServerConfig(t *testing.T) server.Config {
 	gatewayHostAddr := fmt.Sprintf("0.0.0.0:%d", freePort(t))
 	indexRawJSONHostAddr := fmt.Sprintf("0.0.0.0:%d", freePort(t))
 
-	repoPath, err := ioutil.TempDir("/tmp/powergate", ".powergate-*")
-	require.NoError(t, err)
-
+	repoPath := t.TempDir()
```

Comment:
In Go 1.15, we can use `(*testing.T).TempDir()` which automatically cleans the folder when the test finishes, so no need to leave folders in tmp, or return "cls" funcs, and also deletes the folder even if the test panics.
We can use this in lots of places, just changed here.